tests/zenko_tests/README.md

```diff
@@ -137,4 +196,86 @@ $ kubectl get secret mongodb-db-creds -o jsonpath={.data.mongodb-password} | bas
 ```shell
 $ kubectl port-forward dev-db-mongodb-primary-0 27021:27017
 ```
-Connect to `localhost:27021` with database `admin` and `username/password` got from above using your local MongoDB GUI (Robo3T, MongoDB Compass, etc...)
\ No newline at end of file
+Connect to `localhost:27021` with database `admin` and `username/password` got from above using your local MongoDB GUI (Robo3T, MongoDB Compass, etc...)
+
+# How to run zenko end2end test locally with cloudserver and vault
+
+## Prerequisite
+
+- docker
+- redis
+- mongodb image from scality/ci-mongo
+- A local Vault cloned repository with your ongoing modifications
+
+## Set up mongodb
+```shell
+$ docker run -d --net=host --name ci-mongo scality/ci-mongo
+```
+
+### Set up keycloak
+
+First, cd to the Vault repository
+```shell
+$ cd <vault_repository_folder>/.github/docker/keycloak
+```
+
+Then build your Keycloak image:
+```shell
+$ docker build -t keycloak .
+```
+
+Create a configuration file for Keycloak:
+```shell
+$ cat <<EOF > env.list
+KEYCLOAK_REALM=myrealm
+KEYCLOAK_CLIENT_ID=myclient
+KEYCLOAK_USERNAME=bartsimpson
+KEYCLOAK_PASSWORD=123
+KEYCLOAK_USER_FIRSTNAME=Bart
+KEYCLOAK_USER_LASTNAME=Simpson
+EOF
+```
+
+Finally, you can run your keycloak image locally:
+```shell
+$ docker run -p 8443:8443 -p 8080:8080 --env-file env.list -it -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin  keycloak
+```
+
+### Run clouserver
+
+```shell
+$ S3METADATA=mongodb REMOTE_MANAGEMENT_DISABLE=1 S3BACKEND=mem S3VAULT=multiple node index.js
+```
+
+### Run Vault
+
+```shell
+$ VAULT_DB_BACKEND=MONGODB yarn start
+```
+
+### Generate account and account access key using vaultclient
+
+Under vault root folder
+```shell
+$ ADMIN_ACCESS_KEY_ID="D4IT2AWSB588GO5J9T00" ADMIN_SECRET_ACCESS_KEY="UEEu8tYlsOGGrgf4DAiSZD6apVNPUWqRiPG0nTB6" ./node_modules/vaultclient/bin/vaultclient create-account -name account --email acc@ount.fr --port 8600
+$ ADMIN_ACCESS_KEY_ID="D4IT2AWSB588GO5J9T00" ADMIN_SECRET_ACCESS_KEY="UEEu8tYlsOGGrgf4DAiSZD6apVNPUWqRiPG0nTB6" ./node_modules/vaultclient/bin/vaultclient generate-account-access-key --name account --port 8600
+```
+
+### Run end2end tests
+
+```shell
+$ KEYCLOAK_TEST_HOST=http://localhost \
```

Comment:
Maybe adding a quick note on how to manage the configuration in Vault so that Vault successfully connects to Keycloak would help?
->

```json
"jwks": {
    "interval": 300,
    "issuer": "https://localhost:8443/auth/realms/myrealm"
},
"oidcProvider": "https://localhost:8443/auth/realms/myrealm"
```

And make sure the same domain is used in CloudServer (otherwise, if the token `iss` claim contains `127.0.0.1` while Vault only accepts `localhost`, it won't work).